Initial CI Pipeline Setup and golangci-lint Config Fix

arthurbdiniz opened the pull request:
This PR introduces the initial Continuous Integration (CI) pipeline using GitHub Actions for automated testing, formatting, linting, and building of the project.

### 🔧 What's included

- New GitHub Actions workflow (`.github/workflows/ci.yml`):
  - Format step using `gofmt`
  - Lint step via `golangci-lint` (latest version, using GitHub Action)
  - Dependency installation
  - Build step with `go build`
  - Test job running unit tests with coverage
  - TestAcc job for acceptance testing with `TF_ACC=1`

---

### 🛠 Configuration Fix

- Removed invalid `max-per-linter` field from `.golangci.yml`
    - This field has been deprecated and is no longer supported in recent versions (v1.64+).

---

### 📎 References

    [golangci-lint changelog](https://github.com/golangci/golangci-lint/releases)
    [GitHub Actions documentation](https://docs.github.com/en/actions)

---

### 🔒 Signed-off-by

    Arthur Diniz `<arthurbdiniz@gmail.com>`

arthurbdiniz committed:
Initial CI

Signed-off-by: Arthur Diniz <arthurbdiniz@gmail.com>

sohflp commented:
@arthurbdiniz could we apply some branch protection rules to `main` as part of this change?

A few suggestions:
1. Block direct changes to `main` branch, only accept code merge from PR.
2. All pending conversations must be resolved before merge.
3. At least, one person must approve the PR.

arthurbdiniz committed:
Remove max-per-linter from golangci-lint

max-per-linter is not a valid field in recent versions of golangci-lint,
including v1.64+.

That field was removed in older versions (likely pre-1.42).

Signed-off-by: Arthur Diniz <arthurbdiniz@gmail.com>

arthurbdiniz commented:
> @arthurbdiniz could we apply some branch protection rules to `main` as part of this change?
> 
> A few suggestions:
> 
>     1. Block direct changes to `main` branch, only accept code merge from PR.
> 
>     2. All pending conversations must be resolved before merge.
> 
>     3. At least, one person must approve the PR.

Applied all rules suggested here.

Thanks!

arthurbdiniz commented:
Ready for review.

cc @sohflp

sohflp reviewed (approved):
Looks good!
